### PR TITLE
feat(init-capital): add lsd api3 oracle

### PR DESCRIPTION
**Pull request by mewpawee**
- support lsd api3 oracle
- pass registry to subAction validate function on action run

### CODE REVIEW

File: packages/modules/init-capital/src/actions/subactions/setNewPoolOracleReader.ts [hunk 1]
```diff
@@ -119,13 +138,38 @@ export class SetNewPoolOracleReaderSubAction extends SubAction<SetNewPoolOracleR
...
+          if (!registry.api3ProxyOracleReaderProxy) throw new ValidateInputValueError('registry: api3ProxyOracleReader not found')
+          const api3ProxyOracleReaderArtifact = await readArtifact('Api3ProxyOracleReader')
+          const [dataFeedProxy, maxStaleTime] = await this.client.publicClient.readContract({
+            address: registry.api3ProxyOracleReaderProxy,
+            abi: api3ProxyOracleReaderArtifact.abi,
+            functionName: 'dataFeedInfos',
+            args: [params.quoteToken.token],
+          })
+          // check if token parameters has been set
+          if (maxStaleTime === 0n && dataFeedProxy === zeroAddress)
```
Comment (tapakornl) [inline packages/modules/init-capital/src/actions/subactions/setNewPoolOracleReader.ts:170]: should we check `if (maxStaleTime === 0n || dataFeedProxy === zeroAddress)`

File: packages/modules/init-capital/src/actions/subactions/setNewPoolOracleReader.ts [hunk 2]
```diff
@@ -152,12 +198,14 @@ export class SetNewPoolOracleReaderSubAction extends SubAction<SetNewPoolOracleR
       }
       case 'pyth': {
         const params = oracle.params as PythParams
+        // 1. set max staletimes
```
Comment (tapakornl) [inline packages/modules/init-capital/src/actions/subactions/setNewPoolOracleReader.ts:201]: should it be set price ids?

File: packages/modules/init-capital/src/actions/subactions/setNewPoolOracleReader.ts (hunk 1 above)
Comment (mewpawee) [reply packages/modules/init-capital/src/actions/subactions/setNewPoolOracleReader.ts:170]: the comment was wrong

File: packages/modules/init-capital/src/actions/setLsdApi3ProxyOracleReaderTokensInfo.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    super(SetLsdApi3ProxyOracleReaderTokensInfoAction.name, data)
+  }
+
+  protected getSubActions(registry: InitCapitalRegistry): SubAction[] {
+    const governor = this.data.signer['governor']
+    const tokenInfos = this.data.params.tokensInfo
+    const tokens = tokenInfos.map((tokenInfo) => tokenInfo.token)
+    const dataFeedProxies = tokenInfos.map((tokenInfo) => tokenInfo.dataFeedProxy)
+    const maxStaleTimes = tokenInfos.map((tokenInfo) => tokenInfo.maxStaleTime)
+    const quoteTokens = tokenInfos.map((tokenInfo) => tokenInfo.quoteToken)
```
[the remark this answers is not in the document]
Comment (mewpawee) [reply packages/modules/init-capital/src/actions/setLsdApi3ProxyOracleReaderTokensInfo.ts:66]: getAddress already enforce at tx-builder level

**Review by tapakornl — APPROVED** [review]

**Pull request merged**